app/models/user.rb

```diff
@@ -0,0 +1,22 @@
+class User < ActiveRecord::Base
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable and :omniauthable
+  devise :database_authenticatable, :registerable,
+         :recoverable, :rememberable, :trackable, :validatable,
+         :omniauthable, :omniauth_providers => [:facebook]
+
+  has_attached_file :avatar, styles: { medium: "300x300>", thumb: "100x100>" }, default_url: "missing.png"
+  validates_attachment_content_type :avatar, content_type: /\Aimage\/.*\Z/
+
+  has_many :pictures, dependent: :destroy
+  has_many :comments
+
+  def self.from_omniauth(auth)
```

Comment:
Metrics/AbcSize: Assignment Branch Condition size for from_omniauth is too high. [16/15] (http://c2.com/cgi/wiki?AbcMetric)